src/gui/vector/qgsvectorlayerproperties.cpp

```diff
@@ -560,6 +590,22 @@ void QgsVectorLayerProperties::syncToLayer()
   mSimplifyDrawingSpinBox->setValue( simplifyMethod.threshold() );
   mSimplifyDrawingSpinBox->setClearValue( 1.0 );
 
+  QgsVectorLayerSelectionProperties *selectionProperties = qobject_cast< QgsVectorLayerSelectionProperties *>( mLayer->selectionProperties() );
+  if ( QgsSymbol *symbol = selectionProperties->selectionSymbol() )
```

Comment:
Hm, I think it'd be nicer if we wouldn't lose the selection symbol when switching to color override (and vice versa). Could we add a QgsVectorLayerSelectionProperties::{selectionOverrideMode,setSelectionOverrideMode}( None,Color,Symbol ) pair of functions so we can easily switch between modes without losing custom symbol / color?

Reply:
Ok, done!